internal/prober/prober_test.go

```diff
@@ -84,10 +88,28 @@ func TestAPIServerProbeFailure(t *testing.T) {
 
 func TestSuccessfulProbesShouldRunScaleUp(t *testing.T) {
 	leaseList := createNodeLeases([]metav1.MicroTime{nonExpiredLeaseRenewTime, nonExpiredLeaseRenewTime, expiredLeaseRenewTime, expiredLeaseRenewTime})
+	nodeList := createNodes(len(leaseList.Items))
 
 	testCases := []probeTestCase{
-		{name: "Scale Up Succeeds", leaseList: leaseList, minScaleUpCount: 1, maxScaleUpCount: math.MaxInt8},
-		{name: "Scale Up Fails", leaseList: leaseList, scaleUpError: errors.New("scale Up failed"), minScaleUpCount: 1, maxScaleUpCount: math.MaxInt8},
+		{name: "Scale Up Succeeds", leaseList: leaseList, nodeList: nodeList, minScaleUpCount: 1, maxScaleUpCount: math.MaxInt8},
+		{name: "Scale Up Fails", leaseList: leaseList, nodeList: nodeList, scaleUpError: errors.New("scale Up failed"), minScaleUpCount: 1, maxScaleUpCount: math.MaxInt8},
+	}
+
+	for _, entry := range testCases {
+		t.Run(entry.name, func(t *testing.T) {
+			mocks := createAndInitializeMocks(t, entry)
+			config := createConfig(testProbeInterval, metav1.Duration{Duration: time.Microsecond}, metav1.Duration{Duration: 40 * time.Second}, 0.2)
+			createAndRunProber(t, testProbeInterval.Duration, config, mocks)
+		})
+	}
+}
+
+func TestSuccessfulProbesShouldRunScaleUpWithUnrelatedLeases(t *testing.T) {
```

Comment:
```suggestion
func TestLeaseProbeShouldNotConsiderUnrelatedLeases(t *testing.T) {
```
Can you add a case for scale-down in this test itself?